### PR TITLE
PLAT-121900: Fix to close expanded dropdown with back(escape) key

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Issue: Expanded dropdown is not closed on Escape key press.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Solution: Added `escKeyPressed` function and `handleKeyDown` handler to forward onClick on Toggleable. Pressing Escape closes the dropdown.

### Links
[//]: # (Related issues, references)
PLAT-121900

### Comments

### CODE REVIEW

**Review by daniel-stoian-lgp — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ The following is a curated list of changes in the Enact agate module, newest cha
 - `agate/Button` to marquee when focused
 - `agate/Button` to show a tooltip when hovered
 - `agate/DateTimePicker` returned value by onChange event
+- `agate/Dropdown` to support close expanded dropdown with 5-way key
```
Comment (MikyungKim) [inline CHANGELOG.md:23]:
```suggestion
- `agate/Dropdown` to support closing dropdown with back key
```

File: Dropdown/Dropdown.js [hunk 2]
```diff
@@ -365,23 +366,35 @@ const DropDownExtended = hoc((config, Wrapped) => {
...
 		}
 
 		componentWillUnmount () {
 			off('click', this.handleClick);
+			off('keydown', this.handleKeyDown);
 		}
 
 		clickedOutsideDropdown = ({target}) => !this.node.contains(target);
 
+		escKeyPressed = (event) => {
```
Comment (MikyungKim) [inline Dropdown/Dropdown.js:383]: We have `forKey` from `@enact/core/handle` for this purpose.

File: Dropdown/Dropdown.js [hunk 3]
```diff
@@ -365,23 +366,35 @@ const DropDownExtended = hoc((config, Wrapped) => {
...
+		};
+
 		// If a click happened outside the component area close the dropdown by forwarding the onClick from Toggleable.
 		handleClick = handle(
 			this.clickedOutsideDropdown,
 			forward('onClick')
 		).bindAs(this, 'handleClick');
 
+		handleKeyDown = handle(
+			this.escKeyPressed,
```
Comment (MikyungKim) [inline Dropdown/Dropdown.js:394]: Let's `import {forKey} from '@enact/core/handle';` and replace this to `forKey('cancel')`.

Comment (MikyungKim) [reply Dropdown/Dropdown.js:394]: Also, we need to add
```
// Add keymap for escape key
add('cancel', 27);
```
above line 63 after importing `add` like `import {add} from '@enact/core/keymap';`

**Review by MikyungKim — CHANGES_REQUESTED** [review]
Good work. some clean up needed.

**Review by MikyungKim — APPROVED** [review]
Lint, unit, and ui tests are passed locally.
:lgtm:

**Pull request merged**